pandas/core/style.py

```diff
@@ -144,7 +151,18 @@ def __init__(self, data, precision=None, table_styles=None, uuid=None,
         self.table_styles = table_styles
         self.caption = caption
         if precision is None:
-            precision = pd.options.display.precision
+            precision = {'__default__': pd.options.display.precision}
+        elif isinstance(precision, numbers.Integral):
+            precision = {'__default__': pd.options.display.precision}
+        elif isinstance(precision, dict):
+            precision = {k: precision[k] for k in precision} # prevent tampering
+            if '__default__' not in precision:
+                precision['__default__'] = pd.options.display.precision
+        else:
```

Comment:
I would *alway* set the precision to the default (or if a number is passed override to that). Then you can further set precision BY column with another call.